release-notes/supported-os.json

```diff
@@ -0,0 +1,107 @@
+{
+    "$schema": "https://json.schemastore.org/dotnet-supported-os.json",
+    "operating-systems": [
+        {
+            "name": "Ubuntu",
+            "lifecycle-policy": "https://wiki.ubuntu.com/Releases",
+            "last-updated": "2023-02-28",
+            "upcoming-eos": {
+                "version": "23.10",
+                "eos-date": "2024-07-01"
+            },
+            "versions": [
```

Comment:
The `versions` list is technically not necessary, as those informations can be seen in the version-specific files. But having them would allow users to look it up both ways:
* What OS + version is currently supported by a specific .NET version?
* What .NET version is currently supported by a spcific OS + version?

I know that rich definetly wants to have the first option (seen in the version-specific files), but I also like these ones.

Reply:
I think this makes sense if we accompany it with some useful `jq` queries that show how one might take advantage of this data.

An easy one would be "which .NET versions does Ubuntu 22.04 support?" Other examples?

Reply:
The versions mostly have two use cases:
* Can I upgrade .NET when staying on Ubuntu 22.04? (answer: yes, up to .NET 9)
* Can I upgrade from Ubuntu 22.04 to 23.04 when i.e. using .NET 8? (answer: no, only .NET 6)